modules/light-clients/06-solomachine/misbehaviour_handle.go

```diff
@@ -7,8 +7,34 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	commitmenttypes "github.com/cosmos/ibc-go/v6/modules/core/23-commitment/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
 )
 
+// CheckForMisbehaviour returns true for type Misbehaviour (passed VerifyClientMessage check), otherwise returns false
+func (cs ClientState) CheckForMisbehaviour(_ sdk.Context, _ codec.BinaryCodec, _ sdk.KVStore, clientMsg exported.ClientMessage) bool {
+	if _, ok := clientMsg.(*Misbehaviour); ok {
+		return true
+	}
+
+	return false
+}
+
+func (cs ClientState) verifyMisbehaviour(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, misbehaviour *Misbehaviour) error {
+	// NOTE: a check that the misbehaviour message data are not equal is done by
+	// misbehaviour.ValidateBasic which is called by the 02-client keeper.
+	// verify first signature
+	if err := cs.verifySignatureAndData(cdc, misbehaviour, misbehaviour.SignatureOne); err != nil {
+		return sdkerrors.Wrap(err, "failed to verify signature one")
+	}
+
+	// verify second signature
+	if err := cs.verifySignatureAndData(cdc, misbehaviour, misbehaviour.SignatureTwo); err != nil {
+		return sdkerrors.Wrap(err, "failed to verify signature two")
+	}
+
+	return nil
+}
```

Comment:
I think this can go above CheckForMisbehaviour, it will actually be called first in `verifyClientMessage`

Reply:
i checked golang conventions and most commenters seemed to suggest having public functions before private functions. but i am not strongly opinionated here, so if you prefer this first order i can update this PR.

Reply:
My reasoning is that `verifyMisbehaviour` derives from [`VerifyClientMessage`](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/update.go#L28) which is public. It is expected that calls to `VerifyClientMessage`, `CheckForMisbehaviour`, `UpdateState`, `UpdateStateOnMisbehaviour` are performed [together](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/client.go#L65). Thus when I look through light client code, I would expect to review `verifyMisbehaviour` before `CheckForMisbehaviour` because logically the first is required to be correct/called for the second to make sense. I'm not sure the go conventions make sense in this situation since I do not expect nor suggest non-core IBC applications to call these public functions. I don't have strong preference. Happy to let y'all decide, it can always be changed at any time in the future